dump/dump.go

```diff
@@ -113,14 +113,17 @@ func getConsumerConfiguration(ctx context.Context, group *errgroup.Group,
 		return nil
 	})
 
-	group.Go(func() error {
-		oauth2Creds, err := GetAllOauth2Creds(ctx, client, config.SelectorTags)
-		if err != nil {
-			return fmt.Errorf("oauth2: %w", err)
-		}
-		state.Oauth2Creds = oauth2Creds
-		return nil
-	})
+	// konnect doesn't support oauth2, let's skip it
+	if config.KonnectRuntimeGroup == "" {
+		group.Go(func() error {
+			oauth2Creds, err := GetAllOauth2Creds(ctx, client, config.SelectorTags)
+			if err != nil {
+				return fmt.Errorf("oauth2: %w", err)
+			}
+			state.Oauth2Creds = oauth2Creds
+			return nil
+		})
+	}
```

Comment:
It does, but apparently we had a long-lived bug :\ just fixed it here https://github.com/Kong/deck/pull/651/commits/e2c37ee43bd959afeca984462da3b50792d5f5e8

Reply:
Interestingly, this unveiled a nice amount of hidden issues with the integration tests.
If you prefer I can open a new PR only for this

Reply:
> It does, but apparently we had a long-lived bug :\ just fixed it here https://github.com/Kong/deck/commit/e2c37ee43bd959afeca984462da3b50792d5f5e8

That is not a bug. decK intentionally ignores 404 for plugin-entities since plugins could be disabled.

Reply:
figured, I removed the commit

Reply:
Next question, why is this patch required at all? What is the problem with letting the code make the request and receive a 404 response?

Reply:
It seems it isn't. It was just me being confused about the reasons behind some choices. Closing this.